lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -168,11 +168,11 @@ def commit(action, post, account = 'ws', password = @options[:password])
           return Response.new(false, 'Invalid credentials', {}, :test => test?)
         when '403'
           return Response.new(false, 'Not allowed', {}, :test => test?)
-        when '422'
-          return Response.new(false, 'Unprocessable Entity', {}, :test => test?)
-        when '500'
-          if e.response.body.split(' ')[0] == 'validation'
-            return Response.new(false, e.response.body.split(' ', 3)[2], {}, :test => test?)
+        when '422', '500'
+          if e.response.body.split(/\W+/).any? { |word| word  == 'validation' || 'configuration' || 'security' }
+            error_message = e.response.body[/#{Regexp.escape('message=')}(.*?)#{Regexp.escape('&')}/m, 1].tr('+', ' ')
```

Comment:
Wait, how does this work? I thought `.body` was a string; if it is, then how is that regex getting converted into an integer for offset purposes?

Reply:
http://ruby-doc.org/core-2.5.3/String.html#method-i-5B-5D

Reply:
I really hate this language sometimes.